clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/deploy/ops/CreateServerGroupAtomicOperation.java

```diff
@@ -48,50 +50,88 @@ private static Task getTask() {
+      String targetGroupName = description.getTargetGroups().get(0);  // TODO - make target group a single value field in Deck instead of an array here
+      DescribeTargetGroupsResult describeTargetGroupsResult = loadBalancingV2.describeTargetGroups(new DescribeTargetGroupsRequest().withNames(targetGroupName));
+      loadBalancer.setTargetGroupArn(describeTargetGroupsResult.getTargetGroups().get(0).getTargetGroupArn());
     }
-    loadBalancer.setTargetGroupArn(targetGroupARN);
 
     Collection<LoadBalancer> loadBalancers = new LinkedList<>();
     loadBalancers.add(loadBalancer);
 
+
+
     CreateServiceRequest request = new CreateServiceRequest();
-    request.setServiceName(familyName + "-" + APP_VERSION);
-    request.setDesiredCount(description.getDesiredCount() != null ? description.getDesiredCount() : 0);
+    request.setServiceName(serviceName);
+    request.setDesiredCount(description.getCapacity().getDesired()!= null ? description.getCapacity().getDesired() : 1);
     request.setCluster(description.getEcsClusterName());
     request.setRole(description.getIamRole());
     request.setDeploymentConfiguration(deploymentConfiguration);
     request.setLoadBalancers(loadBalancers);
     request.setTaskDefinition(taskDefinition.getTaskDefinitionArn());
 
 
-    getTask().updateStatus(BASE_PHASE, "Creating " + description.getDesiredCount() + " of " + familyName +
+    getTask().updateStatus(BASE_PHASE, "Creating " + description.getCapacity().getDesired() + " of " + familyName +
       " with " + taskDefinition.getTaskDefinitionArn() + " for " + description.getCredentialAccount() + ".");
     ecs.createService(request);
-    getTask().updateStatus(BASE_PHASE, "Done creating " + description.getDesiredCount() + " of " + familyName +
+    getTask().updateStatus(BASE_PHASE, "Done creating " + description.getCapacity().getDesired() + " of " + familyName +
       " with " + taskDefinition.getTaskDefinitionArn() + " for " + description.getCredentialAccount() + ".");
 
-    return null;
+    String serverGroupName = region + ":" + serviceName;  // See in Orca MonitorKatoTask#getServerGroupNames for a reason for this
+
+    DeploymentResult result = new DeploymentResult();
+    result.setServerGroupNames(Arrays.asList(serverGroupName));
+    Map<String, String> namesByRegion = new HashMap<>();
+    namesByRegion.put(region, serviceName);
+
+    result.setServerGroupNameByRegion(namesByRegion);
+
+    return result;
   }
 
-  private TaskDefinition registerTaskDefinition(AmazonECS ecs) {
+  private String inferServergroupVersion() {
+    if (description.getSource() == null) {
+      return "v0001";
+    } else {
+      String[] splitName = description.getSource().getServerGroupName().split("-");
+      String lastPortionOfSplitName = splitName[splitName.length - 1];
```

Comment:
The above two lines can be replaced with:
```
String versionString = StringUtils.substringAfterLast(description.getSource().getServerGroupName(), "-");
```

Apache StringUtils lang3.

Optional.